src/coreclr/src/jit/codegencommon.cpp

```diff
@@ -11466,9 +11466,19 @@ void CodeGenInterface::VariableLiveKeeper::VariableLiveDescriptor::startLiveRang
     // Is the first "VariableLiveRange" or the previous one has been closed so its "m_EndEmitLocation" is valid
     noway_assert(m_VariableLiveRanges->empty() || m_VariableLiveRanges->back().m_EndEmitLocation.Valid());
 
-    // Creates new live range with invalid end
-    m_VariableLiveRanges->emplace_back(varLocation, emitLocation(), emitLocation());
-    m_VariableLiveRanges->back().m_StartEmitLocation.CaptureLocation(emit);
+    if (!m_VariableLiveRanges->empty() && m_VariableLiveRanges->back().m_EndEmitLocation.IsPreviousIns(emit) &&
+        siVarLoc::Equals(&varLocation, &(m_VariableLiveRanges->back().m_VarLocation)))
+    {
+        // The variable is being born at the exactly same place just one assembly instructtion.
```

Comment:
```suggestion
        // The variable is being born at the exactly same place just one assembly instruction.
```

Reply:
I would rephrase this whole comment as:
```
// The variable is being born just after the instruction at which it died.
// In this case, i.e. an update of the variable's value, we coalesce the live ranges.
```
This is more accurate - it actually does die before being reborn, but because it's an update we can report the contiguous range.